### ISSUE TITLE
how to avoid sudo privileges while using the docker container?

### ISSUE THREAD

**Issue by MostafaYA** [oschwengers/asap#7]
Hi Oliver, 
a bit silly question, how to avoid sudo privileges while using the docker container? I am on contos 7 and do not have sudo rights.  
thanks,

**Comment by oschwengers** [comment]
Hi @MostafaYA , not a silly question at all! 
TL;DR
It's currently not possible. You could try Podman or Singularity but I haven't tried them yet and unfortunately have no experience to guide you through this.

Of course, you always need proper `sudo` permissions to start a `Docker` container. As some users asked if it's possible to have the result files have user affiliations instead of `root` I changed the `asap-docker.sh` script in a way, that you *should* execute it without sudo, so it can automatically extract your user name/group and pass this information into the `Docker` container. 
To execute the container, the script will then prompt for `sudo` permissions just before the container execution. I hope this helps. Maybe you could ask your admin to provide you with a VM so you could execute `ASA³P` as `sudo`?

Best regards!

**Comment by jp589** [comment]
Hello @MostafaYA and @oschwengers,

Perhaps I am a bit late to the party, but I only recently discovered the ASA3P pipeline. I also ran into this same issue because I wanted to run this pipeline on my university's HPC grid, but I don't have root privileges there.

I recently built a Singularity image for the pipeline based on the ASA3P pipeline Dockerfile and made some modifications to several of the pipeline files in order to make the the pipeline run inside a single container. The pipeline successfully ran the example files as well as some of my own files.

If you or anyone reading this is interested in the singularity image, please visit https://github.com/jp589/ASA3P_Singularity/ for files and instructions on how to build the image and update the appropriate files.
